src/code/z_map_exp.c

```diff
@@ -100,23 +112,27 @@ s32 Map_GetBossAreaIndex(PlayState* play) {
 }
 
 /**
- * Returns true if the current scene is a boss scene, false otherwise.
+ * Checks if the current scene is a boss scene.
+ * @return true if the current scene is a boss scene, false otherwise.
  */
-s32 Map_IsInBossArea(PlayState* play) {
+s32 Map_IsInBossScene(PlayState* play) {
     if (Map_GetBossAreaIndex(play) == -1) {
         return false;
     }
 
     return true;
 }
 
-s32 D_801BF5A4[] = {
-    SCENE_22DEKUCITY,
-    SCENE_KOEPONARACE,
-    SCENE_F01,
-};
-
-s32 func_8010A238(PlayState* play) {
+/**
+ * Test if the current scene is in a set of overworld scenes.
+ * @return -1 if not in the set, else returns the mapIndex of the overworld scene
+ */
+s32 Map_GetUnkOverworldAreaIndex(PlayState* play) {
```

Comment:
Tbh, I would probably say just leave this unnamed until we find a good name for the overworld areas

Reply:
I don't like leaving the functions unnamed in this case because I've already looked into the three scenes (Deku Palace, Romani Ranch, Gormon Track) and there's no obvious connection between them that I can think of. I also looked into how `gSaveContext.mapIndex` is used because the only instance where `Map_IsInUnkOverworldArea` and `Map_GetUnkOverworldAreaIndex` is called is to change change mapIndex during Map_Init, but `gSaveContext.mapIndex` is only used when reading or updating the player's current dungeon items.

I just think we can make a decision on this now.

Reply:
Hmm, yeah I don't see any correlation and usage doesn't help either as like you said seems to be just related to dungeon items (wonder if those scenes were going to have keys at one point 🤔). Perhaps just replace `Unk` with `Special`, not sure we are going to come up with anything better anyways.